### PR TITLE
fix(dashboard, react-components, core): rewire `ThresholdSection` to new inte…

### PR DESCRIPTION
**Pull request by square-li**
…rfaces of react components

## Overview
1.Update widgets' interfaces and thresholds section on setting panel for new intefaces of widgets in `@iot-app-kit/react-components`.
2. Remove all usage of `synchro-charts`. Use `@iot-app-kit-visualizations` instead.
chrome-capture-2023-2-21

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/dashboard/src/components/sidePanel/sections/thresholdsSection/thresholdComponent.tsx [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 import { COMPARISON_OPERATOR } from '@synchro-charts/core';
-import { Button, Box, Input, Select, SpaceBetween } from '@cloudscape-design/components';
+import type { InputProps, SelectProps } from '@cloudscape-design/components';
+import { Box, Button, Input, Select, SpaceBetween } from '@cloudscape-design/components';
 
 import { DEFAULT_THRESHOLD_COLOR, OPS_ALLOWED_WITH_STRING } from './defaultValues';
 import ColorPicker from '../../shared/colorPicker';
-import type { FC } from 'react';
-import type { ThresholdValue } from '@synchro-charts/core';
-import type { NonCancelableEventHandler } from '@cloudscape-design/components/internal/events';
```
Comment (square-li) [inline packages/dashboard/src/components/sidePanel/sections/thresholdsSection/thresholdComponent.tsx:10]: Remove usage of internal types

File: packages/dashboard/src/components/sidePanel/sections/thresholdsSection/defaultValues.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { COMPARISON_OPERATOR } from '@synchro-charts/core';
+import { ComparisonOperator } from '@iot-app-kit/core';
 
 export const DEFAULT_THRESHOLD_COLOR = '#D0021B';
-export const OPS_ALLOWED_WITH_STRING: COMPARISON_OPERATOR[] = [COMPARISON_OPERATOR.EQUAL, COMPARISON_OPERATOR.CONTAINS];
+export const OPS_ALLOWED_WITH_STRING: ComparisonOperator[] = ['EQ', 'CONTAINS'];
```
Comment (corteggiano) [inline packages/dashboard/src/components/sidePanel/sections/thresholdsSection/defaultValues.ts:4]: we have a COMPARISON_OPERATOR in core which defines these values :)

File: packages/dashboard/testing/mocks/index.ts [hunk 3]
```diff
@@ -219,7 +219,7 @@ export const generateMockTimeSeriesData = (): TimeSeriesData => {
         name: 'test stream',
         data: [dataPoint1, dataPoint2],
         resolution: 0,
-        dataType: DataType.NUMBER,
+        dataType: 'NUMBER',
```
Comment (corteggiano) [inline packages/dashboard/testing/mocks/index.ts:222]: DATA_TYPE from iot-app-kit/core

File: packages/dashboard/src/components/sidePanel/sections/thresholdsSection/thresholdsSection.tsx [hunk 4]
```diff
@@ -41,94 +44,98 @@ const defaultMessages = {
...
+    });
+  };
 
   const comparisonOptions: SelectProps.Option[] = [
-    { label: '>', value: COMPARISON_OPERATOR.GREATER_THAN },
-    { label: '<', value: COMPARISON_OPERATOR.LESS_THAN },
-    { label: '=', value: COMPARISON_OPERATOR.EQUAL },
-    { label: '>=', value: COMPARISON_OPERATOR.GREATER_THAN_EQUAL },
-    { label: '<=', value: COMPARISON_OPERATOR.LESS_THAN_EQUAL },
+    { label: '>', value: 'GT' },
```
Comment (corteggiano) [inline packages/dashboard/src/components/sidePanel/sections/thresholdsSection/thresholdsSection.tsx:126]: COMPARISON_OPERATOR in iot-app-kit/core

File: packages/dashboard/src/components/sidePanel/sections/thresholdsSection/defaultValues.ts (hunk 2 above)
Comment (square-li) [reply packages/dashboard/src/components/sidePanel/sections/thresholdsSection/defaultValues.ts:4]: Good call. updating it

**Review by corteggiano — APPROVED** [review]

File: packages/dashboard/src/customization/settings.ts [hunk 5]
```diff
@@ -11,17 +13,7 @@ export type ComplexFontSettings = {
...
-  thresholds: {
-    id: string;
-    comparisonOperator: string;
-    comparisonValue: string | boolean | number;
-    label?: string;
-    color?: string;
-  }[];
-  colorAcrossThresholds: boolean;
-};
+export type ThresholdWithId = Threshold & { id: string };
```
Comment (jmbuss) [inline packages/dashboard/src/customization/settings.ts:16]: Would id be useful to add in the Threshold type?

**Review by jmbuss — APPROVED** [review]

**Pull request merged**